cfgmgr/vxlanmgr.cpp

```diff
@@ -409,9 +470,138 @@ bool VxlanMgr::doVxlanTunnelMapCreateTask(const KeyOpFieldsValuesTuple & t)
+    if (!isTunnelActive(vxlanTunnelName))
+    {
+        SWSS_LOG_INFO("Vxlan tunnel %s has not been created", vxlanTunnelName.c_str());
+        // Suspend this message until the vxlan tunnel is created
+        return false;
+    }
+
+    if (!isVlanStateOk(vlan))
+    {
+        SWSS_LOG_INFO("VLAN id is not yet created : %s",vxlanTunnelMapName.c_str());
+        return false;
+    }
+
+    // Check the below condition only after the vxlanmgrd has reached reconcile state
+    // The check to verify the state vxlan table is to take care of back to back 
+    // create and delete of a VTEP object. On deletion of a VTEP object the FRR takes 
+    // some time to remove all the routes and once all the routes are removed, the p2p 
+    // tunnel is also removed. This check waits for all the p2p tunnels which were associated
+    // with the earlier version of the VTEP to be deleted before processing further map entry 
+    // creations.
+    WarmStart::WarmStartState state;
+    WarmStart::getWarmStartState("vxlanmgrd",state);
+    if (state == WarmStart::RECONCILED)
+    {
+        if (m_vxlanTunnelMapCache.empty())
+        {
+            std::vector<std::string> keys;
+            m_stateVxlanTunnelTable.getKeys(keys);
+            if (!keys.empty())
+            { 
+                SWSS_LOG_WARN("State VXLAN tunnel table not yet empty.");
+                return false;
+            }
+        }
+    }
+
+    auto sourceIp = std::find_if(
+        it->second.fvt.begin(),
+        it->second.fvt.end(),
+        [](const FieldValueTuple & fvt){ return fvt.first == SOURCE_IP; });
+    if (sourceIp  == it->second.fvt.end())
+    {
+        SWSS_LOG_DEBUG("Vxlan tunnel %s has no field src_ip", vxlanTunnelName.c_str());
+        return true;
+    }
+    else
+    {
+        src_ip = sourceIp->second;
+    }
+    auto dstIp = std::find_if(
+        it->second.fvt.begin(),
+        it->second.fvt.end(),
+        [](const FieldValueTuple & fvt){ return fvt.first == DST_IP; });
+    if (dstIp  != it->second.fvt.end())
+    {
+        dst_ip = dstIp->second;
+    }
+    else
+    {
+        dst_ip = "";
+    }
+
+    createAppDBTunnelMapTable(t);
+    createVxlanNetdevice(vxlanTunnelName, vni_id, src_ip, dst_ip, vlan_id);
+    
+    std::string vxlan_dev_name;
+    vxlan_dev_name = std::string("") + std::string(vxlanTunnelName) + "-" + std::string(vlan_id);
+
+    MapCache map_entry;
+    map_entry.vxlan_dev_name = vxlan_dev_name;
+    map_entry.vlan = vlan;
+    map_entry.vni_id = vni_id;
+
+    m_vxlanTunnelMapCache[vxlanTunnelMapName] = map_entry;
+    m_vlanMapCache[vlan] = vni_id;
+    m_vniMapCache[vni_id] = vlan;
+    m_vxlanTunnelCache[vxlanTunnelName].vlan_vni_refcnt++;
+
+    //Inform the Vlan Mgr to update the tunnel flags if Arp/Nd Suppression is set.
+    std::string key = "Vlan" + std::string(vlan_id);
+    vector<FieldValueTuple> fvVector;
+    FieldValueTuple s("netdev", vxlan_dev_name);
```

Comment:
Is this newly added? I don't remember seeing this in the previous review, May be missed

Reply:
No was part of the first commit itself.